spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceObject.java

```diff
@@ -114,6 +115,12 @@ private Blob resolve() throws IOException {
 	}
 
 	private Blob createBlob() throws IOException {
+		// Creating the bucket resource will create the bucket if necessary.
+		Resource bucket = getBucket();
+		if (!bucket.exists()) {
+			throw new IOException("Unable to create blob: " + getBlobId().toString()
+					+ ", because the bucket does not exist: " + getBlobId().getBucket());
```

Comment:
Does `getBlobId().getBucket()` have the desired `toString()` effect?

Reply:
Yes, the toString() gets the entire path to the blob , and the getBucket() just gets the bucket name.